### ISSUE TITLE
Make checklines taller by default and allow custom-height checklines

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#5062]
In many tracks with checkline issues, the root cause is that the kart can have the possibility to "jump over" (following a collision usually) or "drive under" a checkline (if there is a slope nearby, for example in Black Forest).

This can be resolved by track designers using multiple checklines of the same group positioned at different heights to make sure a kart can't accidentally go around the checkline, however this is a very inelegant solution.

Taller checklines by default would help in most situations.

However, the option to explicitly increase or decrease the height of a specific checkline would be useful for track creators in some situations and should also be added.

**Comment by qwertychouskie** [comment]
I think the cleanest option would be to have checkobjects, so you have have whatever geometry (1 quad, multiple quads/triangles, etc) is needed to properly cover the checkpoint area.

**Comment by kimden** [comment]
I agree with both, but if only check*lines* are supported, that would probably be alright too, just some tracks need custom directions of checklines which I guess is not supported now.

I'd say the same should apply to goal lines. They are currently infinitely high, which has the issue that on arenas with multiple layers the ball can score in unexpected places, which have to be blocked.

**Comment by Alayan-stk-2** [comment]
Done with https://github.com/supertuxkart/stk-code/commit/c8d6c4a74e0678d84364d3a44d3d799214af4fb6
